src/component.js

```diff
@@ -49,8 +49,11 @@ Component.prototype.setState = function(update, callback) {
 	// Skip update if updater function returned null
 	if (update==null) return;
 
+	// The check for _renderCallbacks is because we can't call
+	// this when we're in the constructor but we should call it in
+	// will mount.
+	if (callback && Array.isArray(this._renderCallbacks)) this._renderCallbacks.push(callback);
 	if (this._vnode) {
-		if (callback) this._renderCallbacks.push(callback);
```

Comment:
What about creating it here if it doesn't exist?

```js
		if (callback) (this._renderCallbacks || (this._renderCallbacks = [])).push(callback);
```

Reply:
That would make constructors work in constructor as well, is this intended?

Reply:
I don't know, haha. It seems intuitive to me that the callback would always be called. It seems like this would call the setState callback after initial render, which seems reasonable. Not sure what React does.